nixos/modules/services/x11/display-managers/lightdm-greeters/slick.nix

```diff
@@ -0,0 +1,376 @@
+        '';
+      };
+
+      onscreenKeyboard = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable the onscreen keyboard.
+        '';
+        apply = boolToString;
+      };
+
+      highContrast = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to use a high contrast theme.
+        '';
+        apply = boolToString;
+      };
+
+      screenReader = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable the screen reader.
+        '';
+        apply = boolToString;
+      };
+
+      playReadySound = mkOption {
+        type = types.str;
+        default = "";
+        description = ''
+          A sound file to play when the greeter is ready.
+        '';
+      };
+
+      hiddenUsers = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          List of usernames that are hidden until a special key combination is hit.
+        '';
+      };
+
+      groupFilter = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          List of groups that users must be part of to be shown (empty list shows all users).
+        '';
+      };
+
+      enableHiDPI = mkOption {
+        type = types.enum [
+          "on"
+          "off"
+          "auto"
+        ];
+        default = "auto";
+        description = ''
+          Whether to enable HiDPI support.
+        '';
+      };
+
+      onlyOnMonitor = mkOption {
+        type = types.str;
+        default = "auto";
+        description = ''
+          The monitor on which to show the login window.
+          -1 means "follow the mouse."
+        '';
+      };
+
+      stretchBackgroundAcrossMonitors = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          hether to stretch the background across multiple monitors.
+        '';
+        apply = boolToString;
+      };
+
+      clockFormat = mkOption {
+        type = types.str;
+        default = "%H:%M";
+        description = ''
+          What clock format to use (e.g., %H:%M or %l:%M %p)
+        '';
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Extra configuration that should be put in the lightdm-slick-greeter.conf
+          configuration file.
+        '';
+      };
+    };
+  };
+
+  config = mkIf (ldmcfg.enable && cfg.enable) {
+    services.xserver.displayManager.lightdm = {
+      greeter = mkDefault {
+        package = pkgs.lightdm_slick_greeter.xgreeters;
+        name = "lightdm-slick-greeter";
+      };
+
+      greeters.gtk.enable = mkDefault false;
+    };
+
+    environment.systemPackages = [
+      theme
+      icons
+      font
+    ];
+
+    environment.etc."lightdm/slick-greeter.conf".text = ''
```

Comment:
Could an INI generator be used instead with a `settings` option? Manually including anything configurable seems like a lot of effort to maintain.

Reply:
I think this is probably better, yeah, but I'm not quite sure which convention to follow, because I don't think the INI generator converts camel case to dashes?  I'll do some research and try to do this though.